src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,483 @@
+	*	courseMetadataIds: list of metadata IDs to consider for inclusion/rejection
+	*	dryrun: collect metadata instead of synching
+	* }
+    */
+	/* constructor(app, stats, logger, data) {
+		super(app, stats, logger, data);
+	} */
+
+	/**
+	 * @see {Syncer#respondsTo}
+	 */
+	static respondsTo(target) {
+		return target === 'webuntis-schoolyear';
+	}
+
+	static params(params, data = {}) {
+		// TODO: check usage of params. params.query, and data
+		
+		const query = (params || {}).query || {};
+		
+		const validData = (
+			['inclusive', 'exclusive'].includes(data.datatype) || (!data.datatype && !data.courseMetadataIds)
+		);
+
+		if (!validData) {
+			return false;
+		}
+		
+		return [{
+			datasourceId: params.datasourceId,
+			datatype: data.datatype,
+			courseMetadataIds: data.courseMetadataIds,
+			dryrun: !("dryrun" in params.query) || params.query.dryrun !== 'false',
+		}];
+	}
+
+	/**
+	* @see {Syncer#steps}
+    *
+	* Perform either fetching from WebUntis and storing metadata into 'webuntisMetadata' or
+	* Creating actual schulcloud 'courses'.
+	*/
+	steps() {
+		this.logInfo(`Running WebUntis School Year Sync.\n`);
+
+		if (this.data.dryrun) {
+			return this.createMetaDataFromWebUntisSteps(this.data);
+		} else {
+			return this.createCoursesFromMetaDataAndWebUntisSteps(this.data);
+		}
+	}
+
+	/**
+	* 
+	*/
+	async createMetaDataFromWebUntisSteps(params) {
+		this.stats.success = false;
+		
+		const config = await this.obtainWebUntisConfig();
+
+		const session = await this.login(config);
+		const metaData = await this.fetchMetaData(session, params);
+		await this.logout(session);
+		await this.storeMetadata(metaData, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	*/
+	async createCoursesFromMetaDataAndWebUntisSteps(params) {
+		this.stats.success = false;
+
+		const config = await this.obtainWebUntisConfig();
+
+		const metaData = await this.obtainMetadata(params);
+		const session = await this.login(config);
+
+		const data = await this.fetchData(session, metaData);
+
+		await this.logout(session);
+
+		await this.createCourses(data, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	getImportConditionEvaluator(params) {
+		if (params.datatype === 'inclusive') {
+			return ((id) => params.courseMetadataIds.includes(id));
+		}
+		if (params.datatype === 'exclusive') {
+			return ((id) => !params.courseMetadataIds.includes(id));
+		}
+		throw new BadRequest('invalid datatype');
+	}
+
+	/**
+	* 
+	*/
+	async obtainWebUntisConfig() {
+		return {
+			url: '[URL]',
+			schoolname: '[schoolname]',
+			user: '[username]',
+			password: '[password]',
+		};
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	async obtainMetadata(params) {
+		const metaData = await this.app.service('webuntisMetadata').find({
+			query: { datasourceId: params.datasourceId },
+		});
+
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} config 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)